app/models/profiles_search.rb

```diff
@@ -32,6 +32,7 @@ def detailed_search_result
     return Profile.none if @query.values.all? &:blank?
     result = Profile
       .where('city ILIKE :city', city: "%#{@query[:city]}%")
+      .where('country ILIKE :country', country: "%#{[:country]}")
```

Comment:
You have to add '@query' to the 'country: "%#{[:country]}"' expression (compare to line 34) 
This means country is selected from all parameters that are part of '@query'